cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/sourcecontrol/PullOperationRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.sourcecontrol;
+
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Request type for {@link PullOperation}.
+ */
+public class PullOperationRequest {
+
+  private final RepositoryConfig repoConfig;
+
+  private final NamespaceId namespace;
+
+  private final Set<AppToPull> apps;
+
+  /**
+   * Default Constructor.
+   *
+   * @param namespace namespace for the operation. Currently, we only support pulling
+   *     applications from a single namespace
+   * @param apps Set of apps to pull. Each {@link AppToPull} includes the application name and
+   *     the current hash in db
+   */
+  public PullOperationRequest(NamespaceId namespace, Set<AppToPull> apps,
+      RepositoryConfig repoConfig) {
+    this.namespace = namespace;
+    this.apps = apps;
+    this.repoConfig = repoConfig;
+  }
+
+  public RepositoryConfig getRepoConfig() {
+    return repoConfig;
+  }
+
+  public NamespaceId getNamespace() {
+    return namespace;
+  }
+
+  public Set<AppToPull> getApps() {
+    return apps;
+  }
+
+  /**
+   * Encapsulates information required for a single application to pull.
+   */
+  public static final class AppToPull {
+
+    private final String application;
+    private final String currentGitHash;
```

Comment:
is the hash the commit hash or something else? If it's the commit hash, shouldn't that be a top level property and not specific to each app?